indexing-service/src/main/java/org/apache/druid/indexing/common/task/AbstractBatchIndexTask.java

```diff
@@ -413,6 +420,10 @@ protected boolean tryTimeChunkLock(TaskActionClient client, List<Interval> inter
       if (lock.isRevoked()) {
         throw new ISE(StringUtils.format("Lock for interval [%s] was revoked.", cur));
       }
+      locksAcquired++;
+      if (maxAllowedLockCount >= 0 && locksAcquired > maxAllowedLockCount) {
```

Comment:
Should we move this above where we submit request for lock, and check against current count + 1? At this point, I think we would have exceeded the maxCount by 1?

Reply:
I was originally thinking to prioritize checking for lock revocation, but I moved it and made the check `>=`